### DISCUSSION TITLE
Can't bumpclosefee and no tx listed in pending balance.

### DISCUSSION THREAD

**Discussion by djmuhlestein** [lightningnetwork/lnd#7534] [Troubleshooting] [answered]
So I closed a channel a while back and due to high mempool fees it is stuck in pending.
> lncli wallet bumpclosefee <chan_pt>  # does nothing.  There is no output in the log.  The command does not print any feedback or result.

So, I also noticed that originally I had a pending balance in my on chain wallet but that has disappeared this week.  I can still see the pending close with
> lncli pendingchannels

Unsure how to move things along and resolve.
close Tx: db22a624750115aa5c315943d5f56d84586c47dd85f1329b781bb6a3bf59a871

env:
> lncli --version
lncli version 0.16.0-beta.rc3 commit=v0.16.0-beta.rc3
os:
debian
> btc --version
Bitcoin Core RPC client version v24.0.1

**Comment by guggero** [comment]
This can happen if the fee negotiation with the peer doesn't succeed. You have two options I think: Try re-connecting to the peer, that sometimes works. If not, then you'll have to force close the channel, unfortunately.

**Reply by djmuhlestein** [reply]
> This can happen if the fee negotiation with the peer doesn't succeed. You have two options I think: Try re-connecting to the peer, that sometimes works. If not, then you'll have to force close the channel, unfortunately.

Thanks for the reply..
The node in question is already listed as a peer.  I have two other channels with that node actually.  Those channels are working fine. (node is online). 
Mempool shows the close transaction still.  Just won't clear due to the low fee.   So is this really a situation where I need to force the channel?  I guess if I need to then so be it. How can I investigate further to know for sure?

**Reply by guggero** [reply]
Ah, I didn't see the transaction in my mempool, but you're right, it's in mempool.space. I think the `bumpclosefee` command doesn't work because your node doesn't have the transaction in the mempool anymore. What you can do is increase the mempool size of your `bitcoind` node (`-maxmempool=512`), restart both bitcoind and lnd, make sure the transaction is in your bitcoind (get the raw transaction hex from mempool.space or somewhere and run `bitcoin-cli sendrawtransaction <tx_hex>`), then try the `bumpfee` command again.
That `bumpfee` command creates a sweep transaction that is handled asynchronously. So you'd need to check the logs to see if it is successful or not.

**Reply by djmuhlestein** [reply]
I'm actually running a pruned bitcoind.  no txindex. I wonder if this is related.

**Reply by djmuhlestein** [reply]
But sendrawtransaction is failing -26, min fee not met.  182 < 587.

**Reply by guggero** [reply]
Pruned shouldn't matter. You still have a full mempool, which is what the error means. You need to increase the mempool size, otherwise transactions with a too low fee are just dropped from it.

**Reply by djmuhlestein** [reply]
Indeed that is the min fee problem.  

Ok, new mempool size and I can resend the transaction.  That solved the issue of my unconfirmed balance not showing up in lnd.  
However, bumpclosefee still had no output and nothing in the logs.  It's like the command doesn't do anything.

**Comment by djmuhlestein** [comment]
Updated to 0.16.0 rc5.  Per @guggero suggestion, have transaction back in my local mempool by increasing the mempool size.  However lncli wallet bumpclosefee still returns nothing (no error) and there is no output in the debug log to suggest anything happened.  Still trying to figure out how to debug why the command isn't doing anything.

**Reply by guggero** [reply]
Can you increase your log level to `debug`? Maybe that will show more. And what does `lncli wallet listsweeps --verbose` say?

**Reply by djmuhlestein** [reply]
So increasing lnd loglevel to debug shows that bumpclosefee at least started to execute. I get a single line:
> [DBG] RPCS: [pendingchannels]

Then nothing else (aside from all the extra DBG chatter from other systems) and no error.  When I call pendingchannels myself, I see the pending close channel.

listsweeps returns a list of transactions that are all confirmed:
> lncli wallet listsweeps --verbose | jq '.Sweeps.TransactionDetails.transactions[] | .amount,.num_confirmations'
-333
2407
-333
2926
-667
6595
-667
6596
96088
7476
147
7618

**Reply by guggero** [reply]
So nothing along the lines of `SWPR: Creating sweep transaction` in the logs after you execute the `bumpclosefee` command?
Ah, sorry, I meant `lncli wallet pendingsweeps` as the other one lists the already completed sweeps.

**Reply by djmuhlestein** [reply]
Yes, no other output.  It's like something inside the command short circuited to the end but failed to print the reason why.  As if I had an invalid input but not one that is error checked so it just does nothing.  (On that note, I tried replacing chan_pt with something else and I do get an error message).

Ok pending sweeps is empty as well:
> lncli wallet pendingsweeps 
{
	"pending_sweeps": []
}

**Comment by djmuhlestein** [comment, marked as answer]
Ok, I solved this by short circuiting bumpclosefee and going straight to the bumpfee tool.  I think bumpclosefee has some error in finding the transaction id or something but I haven't tried to actually debug on my node other than watching logs.

So I just used
> lncli wallet bumpfee --sat_per_vbyte <val> txid:0

I just had to use the txid and verfify the output index on mempool manually.
